cli/bpmetadata/types.go

```diff
@@ -0,0 +1,106 @@
+package bpmetadata
+
+import "sigs.k8s.io/kustomize/kyaml/yaml"
+
+// BlueprintMetadata defines the overall structure for blueprint metadata details
+type BlueprintMetadata struct {
+	Meta yaml.ResourceMeta
+	Spec *BlueprintMetadataSpec `yaml:"spec"`
+}
+
+// BlueprintMetadataSpec defines the spec portion of the blueprint metadata
+type BlueprintMetadataSpec struct {
+	Info       BlueprintInfo
+	Content    BlueprintContent
+	Interfaces BlueprintInterface
+	Security   BlueprintSecurity
+}
+
+// BlueprintInfo defines informational detail for the blueprint
+type BlueprintInfo struct {
+	Title         string               `yaml:"title"`
+	Source        *BlueprintRepoDetail `yaml:"source"`
+	Version       string
+	ActuationTool *BlueprintActuationTool
+	Description   *BlueprintDescription
+	Icon          string
+}
+
+// BlueprintContent defines the detail for blueprint related content such as
+// related documentation, diagrams, examples etc.
+type BlueprintContent struct {
+	Diagrams      []BlueprintDiagram
+	Documentation []BlueprintDocumentation
+	SubBlueprints []BlueprintMiscContent
+	Examples      []BlueprintMiscContent
+}
+
+// BlueprintInterface the input and output variables for the blueprint
+type BlueprintInterface struct {
+	Variables      []BlueprintVariable
+	VariableGroups []BlueprintVariableGroup
+	Outputs        []BlueprintOutputs
+}
+
+// BlueprintSecurity defines the roles required and the assocaited services
```

Comment:
nit: security may not be the right term since services. WDYT about Requirements?

Reply:
Sure I can changed that in the next PR. I'll track this and the comment below in a bug in our backlog